openedx/core/lib/tests/test_edx_api_utils.py

```diff
@@ -1,117 +1,197 @@
-"""Tests covering Api utils."""
+"""Tests covering edX API utilities."""
+import json
 import unittest
 
-from django.conf import settings
 from django.core.cache import cache
-from django.test import TestCase
 import httpretty
 import mock
 from nose.plugins.attrib import attr
 from edx_oauth2_provider.tests.factories import ClientFactory
 from provider.constants import CONFIDENTIAL
-from testfixtures import LogCapture
 
-from openedx.core.djangoapps.credentials.models import CredentialsApiConfig
-from openedx.core.djangoapps.credentials.tests.mixins import CredentialsApiConfigMixin, CredentialsDataMixin
 from openedx.core.djangoapps.programs.models import ProgramsApiConfig
-from openedx.core.djangoapps.programs.tests.mixins import ProgramsApiConfigMixin, ProgramsDataMixin
+from openedx.core.djangoapps.programs.tests.mixins import ProgramsApiConfigMixin
 from openedx.core.djangolib.testing.utils import CacheIsolationTestCase
 from openedx.core.lib.edx_api_utils import get_edx_api_data
 from student.tests.factories import UserFactory
 
 
-LOGGER_NAME = 'openedx.core.lib.edx_api_utils'
+UTILITY_MODULE = 'openedx.core.lib.edx_api_utils'
 
 
 @attr('shard_2')
-class TestApiDataRetrieval(CredentialsApiConfigMixin, CredentialsDataMixin, ProgramsApiConfigMixin, ProgramsDataMixin,
-                           CacheIsolationTestCase):
-    """Test utility for API data retrieval."""
+@httpretty.activate
+class TestGetEdxApiData(ProgramsApiConfigMixin, CacheIsolationTestCase):
+    """Tests for edX API data retrieval utility."""
 
     ENABLED_CACHES = ['default']
 
     def setUp(self):
-        super(TestApiDataRetrieval, self).setUp()
-        ClientFactory(name=CredentialsApiConfig.OAUTH2_CLIENT_NAME, client_type=CONFIDENTIAL)
```

Comment:
These tests only need one config model to verify that the helper function works as expected.